### PR TITLE
[FLINK-22820] Allow storing SUSPENDED jobs in FileExecutionGraphInfoS…

### PR DESCRIPTION
**Pull request by fapaul**

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStoreTest.java [hunk 1]
```diff
@@ -76,7 +76,7 @@
     public void testPut() throws IOException {
         final ExecutionGraphInfo dummyExecutionGraphInfo =
                 new ExecutionGraphInfo(
-                        new ArchivedExecutionGraphBuilder().setState(JobStatus.FINISHED).build());
+                        new ArchivedExecutionGraphBuilder().setState(JobStatus.SUSPENDED).build());
```
Comment (tillrohrmann) [inline flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStoreTest.java:79]: I think that we should add an additional test for the `SUSPENDED` case.

**Review by tillrohrmann — APPROVED** [review]
The change itself looks good. Thanks for creating this PR @fapaul.

I had a comment concerning the old test case and I am wondering whether we should add an integration test which test the shutdown of a standalone cluster (using the `MiniCluster`) on which a job is running. This would have caught the problem in the first place. WDYT?

**Comment by fapaul** [comment]
@flinkbot run azure

**Comment by fapaul** [comment]
@tillrohrmann Thanks for the review. I have added an additional test covering the scenario when the FileExecutionGraphInfoStore is used during cluster termination.

**Pull request merged**